doc/design/vic-machine/configure.md

```diff
@@ -0,0 +1,182 @@
+- Consistent with existing update firewall command format (though the command name might be different)
+
+Cons:
+- We'll need to reinvent option names
+
+  For example, to update client network, user need to specify all options if they want to use static ip. Then in one update command, we'll need all those options. Here is the command looks like ```vic-machine configure bridge-network --port-group value --gateway value --ip value```
+  The existing option name in vic-machine create is ```--client-network value   --client-network-gateway value  --client-network-ip value```, we'll need to remove the prefix cause that is already shown as sub-command name.
+
+Decision: We'll go with the first option
+
+## Show Existing VCH configuration
+Right now, there is no user friendly way to view all configurations of existing VCH. To support update operation, we need to make it easier, otherwise, update only makes things weird.
+
+vic-machine inspect command is available now, just it shows only the execution endpoint of VCH, instead of the configuration. We can extend that command, to support configuration inspection as well.
+
+- vic-machine inspect <--configuration, --conf>
+  show VCH configuration only
+- vic-machine inspect <--execution, -e>
+  show VCH execution endpoint (default behavior)
+
+### Configuration Inspection
+For configuration inspection, two output format should be supported
+
+- yml format
+
+  This should print out all VCH configuration with predefined categories, e.g. general information, storage, network, etc. in user friendly format. Secret information will be hidden or skipped.
+
+Following is sample format:
+
+```
+General:
+	Target: "vc.domain.com"
+	Thumbprint: "98:97:D1:7B:DF:..."
+	OpsUser: "Operator"
+	Name: "VCH-1"
+	Proxies: 
+		- "https://host:port"
+		- "http://host:port"
+Compute:
+	Path: " myCluster/Resources/myRP"
+	CpuLimit: 0 MHz
+	MemoryLimit: 0 MHZ
+Storages:
+	ImageStore: "datastore"
+	VolumeStore:
+		- "ds://datastore/volumes/default": "default"
+		- "nfs://userid@host:port/volumes/": "nfs"
+Networks:
+	BridgeNetwork:
+		PortGroup: "bridge"
+		IpRange: "172.16.0.0/12"
+	ClientNetwork:
+		PortGroup: "client"
+		Gateway: "10.1.0.0/16,10.2.0.0/16:10.0.0.1"
+		...
+	DNSServer:
+		- "8.8.8.8"
+		- "8.8.4.4"
+Certificate:
+	...
+Registries:
+	InsecureRegistries:
+		- "..."
+	RegistryCA:
+		- "ca file"
+Endpoint:
+	Cpu: 1
+	Memory: 2048 MHZ
+	Debug: 3
+	SSHEnabled: true
+```
+
+- Command Option Format
+
+  The yml format output is more readable for VIC admin, but if users want to recreate their vic-machine create or configure command, it's hard to do based on that format, because there is no one on one mapping between the configuration and command options. 
+
+  Here we'd support another option as "vic-machine inspect --cmd --conf", this will print out the vic-machine create command options minus secret information. 
```

Comment:
hmm.. didn't quite get your point. I assume this option is the way to get current configuration and the information can easily be updated a little bit for `vic-machine configure`.
As for the manifest file, it's doable to use inspect command to read and even modify manifest file, is that the point?